libsolidity/interface/CompilerStack.cpp

```diff
@@ -225,11 +226,12 @@ bool CompilerStack::parse()
 	if (m_stackState != SourcesSet)
 		BOOST_THROW_EXCEPTION(CompilerError() << errinfo_comment("Must call parse only after the SourcesSet state."));
 	m_errorReporter.clear();
-	ASTNode::resetID();
```

Comment:
Why isn't this reset anymore here?

Reply:
Because the node counter it was moved inside the `Parser` class which is constructed for each `parse()` / `importAST` call and thus starts fresh and needs no resetting.